### DISCUSSION TITLE
Possible to supply custom Dio instance?

### DISCUSSION THREAD

**Discussion by Codel1417** [ArunPrakashG/wordpress_client#69] [Q&A] [answered]
I was wondering if it's possible to supply a custom Dio instance. I use native_dio_adapter and sentry_dio and would like to hook them in. 

I am open to forking myself to add this but was unsure of it was possible already. I haven't checked the code yet.

**Comment by ArunPrakashG** [comment]
Hi @Codel1417 
Currently, we do not have a means of injecting custom dio clients into the instance. I think this could be a really good addition to the library. i will work on the same and share an update!

**Comment by ArunPrakashG** [comment, marked as answer]
@Codel1417 Hey there again! I have updated the library to allow users to pass custom dio instance from constructor. Updated the documentation [here](https://github.com/ArunPrakashG/wordpress_client/wiki/Usage) as well. Please review and let me know if this solves  your use case!

Additionally, you no longer need to initialize separately. I have moved this stop to the constructor itself.

**Reply by Codel1417** [reply]
Thank You!
